man/mysqldumpslow.1
'\" t
.\"     Title: mysqldumpslow
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.79.1 <http://docbook.sf.net/>
.\"      Date: 06/25/2026
.\"    Manual: MySQL Database System
.\"    Source: MySQL 9.7
.\"  Language: English
.\"
.TH "MYSQLDUMPSLOW" "1" "06/25/2026" "MySQL 9\&.7" "MySQL Database System"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.\" mysqldumpslow
.SH "NAME"
mysqldumpslow \- Summarize slow query log files
.SH "SYNOPSIS"
.HP \w'\fBmysqldumpslow\fR\ 'u
\fBmysqldumpslow\fR
.SH "DESCRIPTION"
.PP
\fBmysqldumpslow\fR
is not currently compatible with Percona extended slow query log format\&. Please use
\fBpt-query-digest\fR
from \fBPercona Toolkit\fR instead (https://www.percona.com/doc/percona-toolkit/2.2/pt-query-digest.html)\&.
.SH "COPYRIGHT"
.br
.PP
Copyright \(co 1997, 2026, Oracle and/or its affiliates.
.br
Copyright \(co 2016, Percona Inc.
.PP
This documentation is free software; you can redistribute it and/or modify it only under the terms of the GNU General Public License as published by the Free Software Foundation; version 2 of the License.
.PP
This documentation is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along with the program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA or see http://www.gnu.org/licenses/.
.sp
.SH "SEE ALSO"
For more information, please refer to the MySQL Reference Manual,
which may already be installed locally and which is also available
online at http://dev.mysql.com/doc/.
.SH AUTHOR
Oracle Corporation (http://dev.mysql.com/).
.br
Percona Inc. (http://www.percona.com/).


ChangeLog
------------------------------------------------------------
revno: 3770 [merge]
committer: Bjorn Munch <bjorn.munch@oracle.com>
branch nick: mysql-5.1.65-release
timestamp: Thu 2012-07-12 10:00:14 +0200
message:
  Merge unpushed changes from 5.1.64-release
    ------------------------------------------------------------
    revno: 3756.1.4
    committer: Kent Boortz <kent.boortz@oracle.com>
    branch nick: mysql-5.1.64-release
    timestamp: Tue 2012-06-26 16:30:15 +0200
    message:
      Solve a linkage problem with "libmysqld" on several Solaris platforms:
      a multiple definition of 'THD::clear_error()' in (at least)
      libmysqld.a(lib_sql.o) and libmysqld.a(libfederated_a-ha_federated.o).
      
      Patch provided by Ramil Kalimullin.
    ------------------------------------------------------------
    revno: 3756.1.3
    committer: Joerg Bruehe <joerg.bruehe@oracle.com>
    branch nick: mysql-5.1.64-release
    timestamp: Thu 2012-06-21 16:26:50 +0200
    message:
      Fixing wrong comment syntax (discovered by Kent)
    ------------------------------------------------------------
    revno: 3756.1.2
    committer: Kent Boortz <kent.boortz@oracle.com>
    branch nick: mysql-5.1.64-release
    timestamp: Wed 2012-06-20 13:10:13 +0200
    message:
      Version for this release build is 5.1.64
    ------------------------------------------------------------
    revno: 3756.1.1 [merge]
    committer: Kent Boortz <kent.boortz@oracle.com>
    branch nick: mysql-5.1.64-release
    timestamp: Wed 2012-06-20 13:06:32 +0200
    message:
      Merge
------------------------------------------------------------
revno: 3769
tags: clone-5.1.65-build
committer: Sujatha Sivakumar <sujatha.sivakumar@oracle.com>
branch nick: Bug11762670_5.1
timestamp: Tue 2012-07-10 18:55:07 +0530
message:
  follow up patch for test script failure for BUG#11762670
------------------------------------------------------------
revno: 3768 [merge]
committer: Andrei Elkin <andrei.elkin@oracle.com>
branch nick: mysql-5.1
timestamp: Tue 2012-07-10 13:51:50 +0300
message:
  merge from  5.1 repo.
    ------------------------------------------------------------
    revno: 3764.1.6
    committer: Bjorn Munch <bjorn.munch@oracle.com>
    branch nick: break-51
    timestamp: Tue 2012-07-10 11:57:24 +0200
    message:
      mysql_client_fw.c was not included in make dist
------------------------------------------------------------
revno: 3767 [merge]
committer: Andrei Elkin <andrei.elkin@oracle.com>
branch nick: mysql-5.1
timestamp: Tue 2012-07-10 13:00:03 +0300
message:
  merge from  5.1 repo.
    ------------------------------------------------------------
    revno: 3764.1.5
    committer: Sujatha Sivakumar <sujatha.sivakumar@oracle.com>
    branch nick: Bug11762670_5.1
    timestamp: Tue 2012-07-10 14:23:17 +0530
    message:
      BUG#11762670:MY_B_WRITE RETURN VALUE IGNORED
      
      Problem:
      =======
      The return value from my_b_write is ignored by: `my_b_write_quoted',
      `my_b_write_bit',`Query_log_event::print_query_header'
      
      Most callers of `my_b_printf' ignore the return value. `log_event.cc' 
      has many calls to it. 
      
      Analysis:
      ========
      `my_b_write' is used to write data into a file. If the write fails it
      sets appropriate error number and error message through my_error()
      function call and sets the IO_CACHE::error == -1.
      `my_b_printf' function is also used to write data into a file, it
      internally invokes my_b_write to do the write operation. Upon
      success it returns number of characters written to file and on error
      it returns -1 and sets the error through my_error() and also sets
      IO_CACHE::error == -1.  Most of the event specific print functions
      for example `Create_file_log_event::print', `Execute_load_log_event::print'
      etc are the ones which make several calls to the above two functions and
      they do not check for the return value after the 'print' call. All the above 
      mentioned abuse cases deal with the client side.
      
      Fix:
      ===
      As part of bug fix a check for IO_CACHE::error == -1 has been added at 
      a very high level after the call to the 'print' function.  There are 
      few more places where the return value of "my_b_write" is ignored
      those are mentioned below.
      
      +++ mysys/mf_iocache2.c    2012-06-04 07:03:15 +0000
      @@ -430,7 +430,8 @@
                 memset(buffz, '0', minimum_width - length2);
               else
                 memset(buffz, ' ', minimum_width - length2);
      -        my_b_write(info, buffz, minimum_width - length2);
      
      +++ sql/log.cc	2012-06-08 09:04:46 +0000
      @@ -2388,7 +2388,12 @@
           {
             end= strxmov(buff, "# administrator command: ", NullS);
             buff_len= (ulong) (end - buff);
      -      my_b_write(&log_file, (uchar*) buff, buff_len);
      
      At these places appropriate return value handlers have been added.
------------------------------------------------------------
revno: 3766 [merge]
committer: Andrei Elkin <andrei.elkin@oracle.com>
branch nick: mysql-5.1
timestamp: Tue 2012-07-10 12:48:23 +0300
message:
  merge from  5.1 repo.
    ------------------------------------------------------------
    revno: 3764.1.4
    committer: Bjorn Munch <bjorn.munch@oracle.com>
    branch nick: break-51
    timestamp: Tue 2012-07-10 10:04:57 +0200
    message:
      mysql_client_test did not build within limbysqld/examples
    ------------------------------------------------------------
    revno: 3764.1.3
    committer: Bjorn Munch <bjorn.munch@oracle.com>
    branch nick: grr-51
    timestamp: Mon 2012-07-09 16:36:50 +0200
    message:
      Fixed compile error in mysql_client_test using gcc
    ------------------------------------------------------------
    revno: 3764.1.2
    committer: Bjorn Munch <bjorn.munch@oracle.com>
    branch nick: rfmct-51

scripts/mysqldumpslow.sh
#!@PERL_PATH@

# Copyright (c) 2000, 2023, Oracle and/or its affiliates.
# Copyright (c) 2016, Percona Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License, version 2.0,
# as published by the Free Software Foundation.
#
# This program is also distributed with certain software (including
# but not limited to OpenSSL) that is licensed under separate terms,
# as designated in a particular file or component or in included license
# documentation.  The authors of MySQL hereby grant you an additional
# permission to link the program and your derivative works with the
# separately licensed software that they have included with MySQL.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License, version 2.0, for more details.
#
# You should have received a copy of the GNU Library General Public
# License along with this library; if not, write to the Free
# Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston,
# MA 02110-1301, USA

# mysqldumpslow - parse and summarize the MySQL slow query log

# Original version by Tim Bunce, sometime in 2000.
# Further changes by Tim Bunce, 8th March 2001.
# Handling of strings with \ and double '' by Monty 11 Aug 2001.

usage();

sub usage {
    my $text= <<HERE;
"mysqldumpslow.sh" is not currently compatible with Percona extended slow query
log format. Please use "pt-query-digest" from Percona Toolkit instead
(https://www.percona.com/doc/percona-toolkit/2.2/pt-query-digest.html).

HERE
    print $text;
    exit 0;
}
